tools/toolchains/gcc.py

```diff
@@ -52,37 +53,48 @@ def __init__(self, target,  notify=None, macros=None, build_profile=None,
             self.flags["common"].append("-DMBED_RTOS_SINGLE_THREAD")
             self.flags["ld"].append("--specs=nano.specs")
 
-        if target.core == "Cortex-M0+":
+        core = target.core
+        if CORE_ARCH[target.core] == 8:
+            # Add linking time preprocessor macro DOMAIN_NS
+            if target.core.endswith("-NS"):
+                self.flags["ld"].append("-DDOMAIN_NS=1")
+                core = target.core[:-3]
+            else:
+                self.cpu.append("-mcmse")
+                self.flags["ld"].extend([
+                    "-Wl,--cmse-implib",
+                    "-Wl,--out-implib=%s" % join(build_dir, "cmse_lib.o")
+                ])
+
+        if core == "Cortex-M0+":
```

Comment:
Could we convert this into a lookup table please?

Reply:
Done